### PR TITLE
Update SDL-0219 with accepted revisions

### PR DESCRIPTION
**Pull request by mvyrko**
**Updates**

Add description of functions for mandatory values

### CODE REVIEW

**Comment by mvyrko** [comment]
@jordynmackool please, review

File: proposals/0219-ios-check-type.md [hunk 1]
```diff
@@ -30,18 +30,33 @@ For example:
...
+
+For mandatory values, next functions try to return object of classType overwise error would be filled. 
+In case, when error isn't `nil`, returned object can be not type of `classType`:
+`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error`
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error;`.
 
 For suitable working with SDLEnums create the following methods:
-`- (nullable SDLEnum)sdl_enumForName:(SDLName)name`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name;`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError *);`.
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
`- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError **)error;
```

File: proposals/0219-ios-check-type.md [hunk 2]
```diff
@@ -30,18 +30,33 @@ For example:
...
 `- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name;`
+`- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name error:(NSError *);`.
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
`- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name error:(NSError **)error;
```

File: proposals/0219-ios-check-type.md [hunk 3]
```diff
@@ -30,18 +30,33 @@ For example:
...
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.
+
+For mandatory values, next functions try to return object of classType overwise error would be filled. 
+In case, when error isn't `nil`, returned object can be not type of `classType`:
+`- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error`
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType error:(NSError **)error;`.
 
 For suitable working with SDLEnums create the following methods:
-`- (nullable SDLEnum)sdl_enumForName:(SDLName)name`
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name;`
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]: Is this needed? it's not included in the requested revisions.

File: proposals/0219-ios-check-type.md [hunk 4]
```diff
@@ -30,18 +30,33 @@ For example:
...
+`- (nullable SDLEnum)sdl_enumForName:(SDLName)name error:(NSError *);`.
 `- (nullable NSArray<SDLEnum> *)sdl_enumsForName:(SDLName)name;`
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]: This can be removed, per the requested revisions.

File: proposals/0219-ios-check-type.md [hunk 5]
```diff
@@ -30,18 +30,33 @@ For example:
...
 ## Proposed solution
 
-Remove method :
+Mark method as private:
 `-(nullable id)sdl_objectForName:(SDLName)name;`. 
 
-Instead of the removed method, use this existing method:
+Instead of method above use next methods:
+
+For optional values, can be used functions that return object of classType or nil:
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
For optional values, can use functions that return object of classType or nil:
```

File: proposals/0219-ios-check-type.md [hunk 6]
```diff
@@ -30,18 +30,33 @@ For example:
...
 `- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;`.
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.
+
+For mandatory values, next functions try to return object of classType overwise error would be filled.
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
For mandatory values, the following functions try to return object of classType overwise error will be filed. 
```

File: proposals/0219-ios-check-type.md [hunk 7]
```diff
@@ -30,18 +30,33 @@ For example:
...
 
-Instead of the removed method, use this existing method:
+Instead of method above use next methods:
+
+For optional values, can be used functions that return object of classType or nil:
 `- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType;`.
+`- (nullable NSArray *)sdl_objectsForName:(SDLName)name ofClass:(Class)classType`.
+
+For mandatory values, next functions try to return object of classType overwise error would be filled. 
+In case, when error isn't `nil`, returned object can be not type of `classType`:
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
When the object is not of type `classType`, the error will not be `nil`
```

File: proposals/0219-ios-check-type.md [hunk 8]
```diff
@@ -60,69 +75,21 @@ Proposal implemented in https://github.com/smartdevicelink/sdl_ios/pull/1158
 
 ## Potential downsides
 
-Store returns `nil` by getting with the wrong key or incorrect type without highlighting the on compile time.
+Store can return wrong type that would be highlighted by error.
```
Comment (theresalech) [inline proposals/0219-ios-check-type.md]:
```suggestion
Store can return wrong type, which would be highlighted by an error.
```

File: proposals/0219-ios-check-type.md (hunk 3 above)
Comment (mvyrko) [reply proposals/0219-ios-check-type.md]: In my view, we can save this method if we need to getting SDLEnum and don't want to handle error.
```
- (nullable SDLEnum)sdl_enumForName:(SDLName)name {
    return [self sdl_objectForName:name ofClass:NSString.class];
}
- (nullable id)sdl_objectForName:(SDLName)name ofClass:(Class)classType {
    NSError *error;
    id object = [self sdl_objectForName:name ofClass:classType error:&error];
    return error == nil ? object : nil;
}
```

File: proposals/0219-ios-check-type.md (hunk 4 above)
Comment (mvyrko) [reply proposals/0219-ios-check-type.md]: This method calls inside method with error, like example above

**Comment by mvyrko** [comment]
@theresalech updated, please take a look again

**Comment by theresalech** [comment]
Hi @mvyrko - regarding your last two comments, we discussed the need for methods not containing an `error` parameter on the Steering Committee call and decided to just include it for every RPC parameter call. So this non-error enum method is not needed and should not be included.  Please remove and I will move forward with merging the PR.  Thank you!

**Comment by mvyrko** [comment]
got it, will do

**Comment by mvyrko** [comment]
Hello @theresalech, enum's methods without NSError parameter were removed in 0d9b8f3.
Please, take a look again

**Pull request merged**